client/starwhale/utils/venv.py

```diff
@@ -107,12 +108,18 @@ def _do_pip_install_req(
     _hosts: t.List[str] = [_env.get("SW_PYPI_TRUSTED_HOST", "")]
     _index: str = _env.get("SW_PYPI_INDEX_URL", "")
 
+    list_to_str: t.Callable[[t.Union[str, list]], str] = (
+        lambda x: " ".join(x) if isinstance(x, list) else x
+    )
+
+    _pip_index_url = list_to_str(pip_config.get("index_url", ""))
```

Comment:
`pip_config.get("index_url", "")` will always returns string?

Reply:
yes， only for mypy check.